File: WebFt30/src/app/core/store/models/monitoring-alarm.ts
import {
  MonitoringAlarmEvent,
  MonitoringAlarmLevel,
  MonitoringAlarmStatus
} from './monitoring-alarm-event';
import { MonitoringAlarmType } from './monitoring-alarm-type';

export class MonitoringAlarm {
  id!: number;
  alarmGroupId!: number;
  monitoringPortId!: number;
  monitoringResultId!: number;
  lastChangedAt!: Date;
  activeAt!: Date;
  resolvedAt!: Date | null;
  type!: MonitoringAlarmType;
  level!: MonitoringAlarmLevel;
  status!: MonitoringAlarmStatus;
  distanceMeters!: number | null;

  events!: MonitoringAlarmEvent[];
}

File: Backend/src/Fibertest30.Application/Alarm/Models/MonitoringAlarm.cs
namespace Fibertest30.Application;

public class MonitoringAlarm
{
    public int Id { get; set; }
    public int AlarmGroupId { get; set; }
    public int MonitoringPortId { get; init; }
    public int MonitoringResultId { get; init; }
    public DateTime LastChangedAt { get; set; }
    public DateTime ActiveAt { get; set; }
    public DateTime? ResolvedAt { get; set; }
    public MonitoringAlarmType Type { get; init; }
    public MonitoringAlarmLevel Level { get; set; }
    public MonitoringAlarmStatus Status { get; set; }
    public double? DistanceMeters { get; set; }
    
    public List<MonitoringAlarmEvent>? Events { get; set; }
}
